File: 0856-score-of-parentheses/0856-score-of-parentheses.py
class Solution:
    def scoreOfParentheses(self, s: str) -> int:
        
        mult = 1
        score = 0
        i = 0
        while i < len(s):
            c = s[i]
            if c == '(':
                if i + 1 < len(s) and s[i + 1] == '(':
                    mult *= 2
                else:
                    score += mult
                    i += 1
            else:
                mult //= 2
            i += 1
        return score

File: 0856-score-of-parentheses/0856-score-of-parentheses.java
class Solution {
    public int scoreOfParentheses(String s) {
        /*
            ; at ending 
            s.charAt(i)
            s.length()
            &&
        */
        int mult = 1;
        int score = 0;
        
        for (int i=0; i<s.length(); i++){
            if (s.charAt(i) == '('){
                if (i+1 < s.length() && s.charAt(i+1) == '(' ){
                    mult *= 2;
                }else{
                    score += mult;
                    i += 1;
                }
            }else{
                mult /= 2;
            }
                
        }
        return score;
    }
}